templates/demo-store/app/data/cache.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  CacheLong,
+  CacheNone,
+  CacheShort,
+  generateCacheControlHeader,
+} from '@shopify/hydrogen';
+
+export function routeHeaders({loaderHeaders}: {loaderHeaders: Headers}) {
+  // Keep the same cache-control headers when loading the page directly
+  // versus when transititioning to the page from other areas in the app
+  return {
+    'Cache-Control': loaderHeaders.get('Cache-Control'),
+  };
+}
+
+export const CACHE_SHORT = generateCacheControlHeader(CacheShort());
+export const CACHE_LONG = generateCacheControlHeader(CacheLong());
+export const CACHE_NONE = generateCacheControlHeader(CacheNone());
```

Comment:
What do you think of doing this instead?

```suggestion
export const HEADERS_CACHE_SHORT = Object.freeze({'Cache-Control': generateCacheControlHeader(CacheShort())});
export const HEADERS_CACHE_LONG = Object.freeze({'Cache-Control': generateCacheControlHeader(CacheLong())});
export const HEADERS_CACHE_NONE = Object.freeze({'Cache-Control': generateCacheControlHeader(CacheNone())});
```

Then, in loaders we can just use `{headers: HEADERS_CACHE_SHORT}` instead of typing "cache-control" everywhere.

Reply:
mmm .. I like what Bret have:

If we go for Fran's suggestion ...
```js
// it's easy when nothing else is needed
json({}, {headers: HEADERS_CACHE_SHORT})

// it's weird when you need to add something else
json({}, {
  headers: {
    ...HEADERS_CACHE_SHORT,
    'some-header': 'value'
  }
})
```

whereas Bret's suggestion would be unambiguous
```js
json({}, {
  headers: {
    'Cache-Control': HEADERS_CACHE_SHORT,
    'some-header': 'value'
  }
})
```